core/src/main/java/org/elasticsearch/action/support/AutoCreateIndex.java

```diff
@@ -67,11 +69,13 @@ public boolean needToCheck() {
     /**
      * Should the index be auto created?
      */
-    public boolean shouldAutoCreate(String index, ClusterState state) {
+    public boolean shouldAutoCreate(String index, ClusterState state, IndexNameExpressionResolver resolver) {
         if (!needToCheck) {
             return false;
         }
-        if (state.metaData().hasConcreteIndex(index)) {
+        IndicesOptions options = IndicesOptions.fromOptions(true, true, true, true);
+        String[] concreteIndices = resolver.concreteIndices(state, options, index);
+        if (concreteIndices.length > 0) {
```

Comment:
maybe it's me but this change seems fishy... we do much more than before now here, also not using indices options coming from the request... but first of all are we sure we want to support writing into index names containing date math expressions? in my mind it was more about reading, search api etc.